tests/ssg_test_suite/profile.py

```diff
@@ -1,74 +1,39 @@
 #!/usr/bin/env python2
 from __future__ import print_function
 
-import atexit
 import logging
-import sys
+
 
 import ssg_test_suite.oscap
 import ssg_test_suite.virt
 from ssg_test_suite.rule import get_viable_profiles
-from ssg_test_suite.virt import SnapshotStack
 
 logging.getLogger(__name__).addHandler(logging.NullHandler())
 
 
-def perform_profile_check(options):
-    """Perform profile check.
+class ProfileChecker(ssg_test_suite.oscap.Checker):
+    def _test_target(self, target):
+        profiles = get_viable_profiles(
```

Comment:
I have restructured the code to be more self-documenting and re-added some of those comments. However, when a comment describes functionality of code that is present in another module, there is a high likelihood that it will become obsolete without anybody noticing.